### DISCUSSION TITLE
Users can't able to see the lists of bucket they have created

### DISCUSSION THREAD

**Discussion by abdulfasithgit** [minio/minio#11663] [Q&A] [answered]
I have policy like this
`
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "s3:CreateBucket",
        "s3:ListAllMyBuckets"
      ],
      "Effect": "Allow",
      "Resource": [
        "arn:aws:s3:::*"
      ],
      "Sid": ""
    },
    {
      "Action": [
        "s3:GetObject",
        "s3:GetBucketLocation",
        "s3:ListBucket"
      ],
      "Effect": "Allow",
      "Resource": [
        "arn:aws:s3:::*"
      ],
      "Sid": ""
    }
  ]
}
`
## Expected Behavior

Users are not able to see the list  of buckets they created via Web minio.

##Steps to Reproduce (for bugs)
1. Added a user.
2. Assigned policy as mentioned above to the user.
3. Logged in with user credentials.
4. Created a bucket

## Your Environment

* Version used (`minio --version`): 2021-02-19T04:38:02Z
* Server setup and configuration:  linux 
* Operating System and version (`uname -a`):Linux localhost.localdomain 4.18.0-240.10.1.el8_3.x86_64 #1 SMP Mon Jan 18 17:05:51 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

**Comment by harshavardhana** [comment, marked as answer]
@abdulfasithgit your policy lists all buckets - and there is no user that owns a bucket - so users don't own buckets they are simply authorized to access buckets.

**Reply by trowdan** [reply]
how can I access (read/write) in the UI only the buckets created by me? Could you suggest some IAM policies to use?

**Reply by harshavardhana** [reply]
"Buckets" do not have any Creation level permissions - they are all controlled by policies so bucket is not really "owned" by someone. It is available for everyone to use, you can restrict their access patterns that's all. 

So for example providing `s3:PutObject, s3:GetObject` with the specific resource `bucketname` gets you as a user access to read and write objects to your bucket.

**Reply by trowdan** [reply]
What if I want to create the following scenario:

- User A creates bucket1, bucket2, bucket3
- User B creates bucket4

- User A cannot see bucket4
- User B creates bucket1, bucket2, bucket3

and I want to do this using a single policy, without changing it time to time when a user create a new bucket

is it possible to do that?

PS: I have integrated my minIO instance with OpenID, I don't know if it's possible to restrict based on some jwt fields

Thank you for your time

**Reply by harshavardhana** [reply]
https://github.com/minio/minio/tree/master/docs/multi-user#policy-variables read here
